src/adams_utils.jl

```diff
@@ -0,0 +1,56 @@
+function ϕ_and_ϕstar!(cache, dy, next_point, last_idx)
+	@unpack grid_points, ϕstar_nm1, k = cache
+	ϕ_n = []
```

Comment:
Allocation, and this is not strictly typed. This will dramatically slow the code down. You should strictly type this and cache the array.